var/spack/repos/builtin/packages/intel-mpi/package.py

```diff
@@ -54,6 +54,13 @@ def mpi_libs(self):
             libraries, root=self.prefix.lib64, shared=True, recurse=True
         )
 
+    @property
+    def mpi_headers(self):
+        # recurse from self.prefix will find too many things for all the
+        # supported sub-architectures like 'mic'
+        return find_headers(
+            'mpi', root='{0}64'.format(self.prefix.include), recurse=False)
```

Comment:
Oh yes, I'm not suggesting you get rid of `mpi_headers`, just saying you could replace `'{0}64'.format(self.prefix.include)` with `self.prefix.include64`.